### ISSUE TITLE
Screen working on Fedora 37

### ISSUE THREAD

**Issue by grillo-delmal** [eleclab-rpi/5INCH_DSI_DISPLAY#3]
I purchased an ElecLab Raspberry Pi Touchscreen Monitor 5 Inch DSI IPS Capacitive LCD Display 800x480 on Amazon a long time ago, and during last week I was toying around to make the touchscreen work on Fedora 37 beta[0] on my rpi4. These are the commands I ran when running the image on my device:

```sh
# This 2 lines enable the Firmware Device Tree instead of the upstream kernel one [1]
# With these the screen works and we can use device overlays
$ sudo rm /boot/dtb
$ sudo echo "FirmwareDT=True" >> /etc/u-boot.conf

# The next line enables the generic rpi touchscreen overlay
$ sudo echo "dtoverlay=rpi-ft5406,touchscreen-size-x=800,touchscreen-size-y=420" >> /boot/efi/config.txt
```

After rebooting, the touchscreen worked without problems. The only thing that doesn't work in this image is the audio jack, but it seems to be expected in Fedora [2].

Are there any side effects that could arise when enabling the touchscreen this way?

[0] https://getfedora.org/workstation/download/
[1] https://fedoraproject.org/wiki/Architectures/ARM/Raspberry_Pi/HATs#General_configuration
[2] https://fedoraproject.org/wiki/Architectures/ARM/Raspberry_Pi#Does_sound_work?

**Comment by eleclab-rpi** [comment]
Dear friends, this is a 5" 480x854 DPI display, the touch screen IC is FT6336U, the parameters are different, the configuration parameters of config.txt in buster/bullseye are as follows:

[all]
#dtoverlay=vc4-fkms-v3d
gpio=0-9=a2
gpio=12-17=a2
gpio=20-25=a2
dtoverlay=dpi24
enable_dpi_lcd=1
display_default_lcd=1
extra_transpose_buffer=2
dpi_group=2
dpi_mode=87
dpi_output_format=0x78216
dpi_timings=480 0 60 2 55 854 0 16 2 22 0 0 0 60 0 32000000 6 #for no rotate or for display_rotate=2
#dpi_timings=480 0 60 2 55 856 0 16 2 22 0 0 0 60 0 32000000 6 #for display_rotate=1,3
dtoverlay=eleclab-dpi5inch
#display_rotate=1 #1:90;2:180;3:270 , Need to modify 40-libinput.conf



***@***.***
 
From: Grillo
Date: 2022-09-29 14:17
To: eleclab-rpi/5INCH_DSI_DISPLAY
CC: Subscribed
Subject: [eleclab-rpi/5INCH_DSI_DISPLAY] Screen working on Fedora 37 (Issue #3)
I purchased an ElecLab Raspberry Pi Touchscreen Monitor 5 Inch DSI IPS Capacitive LCD Display 800x480 on Amazon a long time ago, and during last week I was toying around to make the touchscreen work on Fedora 37 beta[0] on my rpi4. These are the commands I ran when running the image on my device:
# This 2 lines enable the Firmware Device Tree instead of the upstream kernel one [1]
# With these the screen works and we can use device overlays
$ sudo rm /boot/dtb
$ sudo echo "FirmwareDT=True" >> /etc/u-boot.conf
# The next line enables the generic rpi touchscreen overlay
$ sudo echo "dtoverlay=rpi-ft5460,touchscreen-size-x=800,touchscreen-size-y=420" >> /boot/efi/config.txt
After rebooting, the touchscreen worked without problems. The only thing that doesn't work in this image is the audio jack, but it seems to be expected in Fedora [2].
Are there any side effects that could arise when enabling the touchscreen this way?
[0] https://getfedora.org/workstation/download/
[1] https://fedoraproject.org/wiki/Architectures/ARM/Raspberry_Pi/HATs#General_configuration
[2] https://fedoraproject.org/wiki/Architectures/ARM/Raspberry_Pi#Does_sound_work?

**Comment by eleclab-rpi** [comment]
Of course, eleclab-dpi5inch.dtbo should be placed in the /boot/overlays/ directory, the touchscreen won't work without it.



***@***.***
 
From: Grillo
Date: 2022-09-29 14:17
To: eleclab-rpi/5INCH_DSI_DISPLAY
CC: Subscribed
Subject: [eleclab-rpi/5INCH_DSI_DISPLAY] Screen working on Fedora 37 (Issue #3)
I purchased an ElecLab Raspberry Pi Touchscreen Monitor 5 Inch DSI IPS Capacitive LCD Display 800x480 on Amazon a long time ago, and during last week I was toying around to make the touchscreen work on Fedora 37 beta[0] on my rpi4. These are the commands I ran when running the image on my device:
# This 2 lines enable the Firmware Device Tree instead of the upstream kernel one [1]
# With these the screen works and we can use device overlays
$ sudo rm /boot/dtb
$ sudo echo "FirmwareDT=True" >> /etc/u-boot.conf
# The next line enables the generic rpi touchscreen overlay
$ sudo echo "dtoverlay=rpi-ft5460,touchscreen-size-x=800,touchscreen-size-y=420" >> /boot/efi/config.txt
After rebooting, the touchscreen worked without problems. The only thing that doesn't work in this image is the audio jack, but it seems to be expected in Fedora [2].
Are there any side effects that could arise when enabling the touchscreen this way?
[0] https://getfedora.org/workstation/download/
[1] https://fedoraproject.org/wiki/Architectures/ARM/Raspberry_Pi/HATs#General_configuration
[2] https://fedoraproject.org/wiki/Architectures/ARM/Raspberry_Pi#Does_sound_work?
